### PR TITLE
Refactor master list usage

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### Added

- Tests

### Changed

- Refactor master list usage

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b refactor-master-list-usage
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/constants/gene_panel.py [hunk 1]
```diff
@@ -64,6 +64,18 @@ def collaborators() -> set[str]:
...
+    @staticmethod
+    def is_customer_collaborator_and_panels_in_gene_panels_master_list(
+        customer_id: str, gene_panels: set[str]
+    ) -> bool:
+        return customer_id in GenePanelMasterList.collaborators() and gene_panels.issubset(
+            GenePanelMasterList.get_panel_names()
+        )
+
+    @staticmethod
+    def get_broad_non_specific_gene_panels() -> set[str]:
```
Comment (ChrOertlin) [inline cg/constants/gene_panel.py:76]:
```suggestion
    def get_non_specific_gene_panels() -> set[str]:
```
I think broad and non-specific are a tautology here.

**Review by ChrOertlin — APPROVED** [review]

File: cg/constants/gene_panel.py (hunk 1 above)
Comment (henrikstranneheim) [reply cg/constants/gene_panel.py]: Fixed

**Comment by henrikstranneheim** [comment]
###Starting MIP, 👍 
```
[henrik.stranneheim@hasta:~] [S_base] $ cg workflow mip-dna config-case --dry-run justhusky
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Case justhusky exists in Status DB
{'case': 'justhusky', 'default_gene_panels': ['OMIM-AUTO'], 'samples': [{'sample_id': 'earlycasualcaiman', 'sample_display_name': 'ACC5339A2--300M', 'analysis_type': 'wgs', 'sex': 'male', 'phenotype': 'unaffected', 'expected_coverage': 1, 'capture_kit': 'twistexomecomprehensive_10.2_hg19_design.bed', 'mother': '0', 'father': '0'}, {'sample_id': 'slowlycivilbuck', 'sample_display_name': 'ACC5339A3--300M', 'analysis_type': 'wgs', 'sex': 'female', 'phenotype': 'unaffected', 'expected_coverage': 1, 'capture_kit': 'twistexomecomprehensive_10.2_hg19_design.bed', 'mother': '0', 'father': '0'}, {'sample_id': 'hugelymodelbat', 'sample_display_name': 'ACC5339A4--300M', 'analysis_type': 'wgs', 'sex': 'male', 'phenotype': 'affected', 'expected_coverage': 1, 'capture_kit': 'twistexomecomprehensive_10.2_hg19_design.bed', 'mother': 'slowlycivilbuck', 'father': 'earlycasualcaiman'}]}
[henrik.stranneheim@hasta:~] [S_base] $ cg workflow mip-dna start --dry-run justhusky
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Case justhusky exists in Status DB
Starting full MIP analysis workflow for case justhusky
Flow cell check is not applicable - the case is either down sampled or external.
Found file earlycasualcaiman/2019-05-08/HHT5NDSXX_HapMap_S26_L001_R1_001.fastq.gz
Found file earlycasualcaiman/2019-05-08/HHT5NDSXX_HapMap_S26_L001_R2_001.fastq.gz
Found file slowlycivilbuck/2019-05-08/HHT5NDSXX_HapMap_S27_L001_R1_001.fastq.gz
Found file slowlycivilbuck/2019-05-08/HHT5NDSXX_HapMap_S27_L001_R2_001.fastq.gz
Found file hugelymodelbat/2019-05-08/HHT5NDSXX_HapMap_S28_L001_R1_001.fastq.gz
Found file hugelymodelbat/2019-05-08/HHT5NDSXX_HapMap_S28_L001_R2_001.fastq.gz
Case justhusky exists in Status DB
Case justhusky exists in Status DB
Running command /home/proj/stage/bin/miniconda3/envs/S_scout/bin/scout --config /home/proj/stage/servers/config/hasta.scilifelab.se/scout-stage.yaml export panel --bed BRAIN Cardiology CILM CH CTD DIAB ENDO EP HEARING HYDRO IBMFS IEM IF mcarta MHT MIT MOVE mtDNA NBS-M NEURODEG NMD OMIM-AUTO OPTIC PANELAPP-GREEN PEDHEP PID PIDCAD RETINA SKD SOVM STROKE AID Inherited cancer --build 37

**Comment by henrikstranneheim** [comment]
Deployed in production:
```
Log deploy... done.
cg, version 61.3.2

**Pull request merged**